src/idpi/grib_decoder.py

```diff
@@ -309,3 +314,36 @@ def load_fieldnames(
     ) -> dict[str, xr.DataArray]:
         reqs = {param: param for param in params}
         return self.load(reqs, extract_pv)
+
+
+def _get_type_of_level(field):
+    if field.vcoord_type == "model_level":
+        if field.origin["z"] == 0.0:
+            return "generalVerticalLayer"
+        elif field.origin["z"] == -0.5:
+            return "generalVertical"
+        else:
+            raise ValueError(f"Unsupported field origin in z: {field.origin['z']}")
+    else:
+        mapping = {vc: name for name, (vc, _) in VCOORD_TYPE.items()}
+        return mapping.get(field.vcoord_type, field.vcoord_type)
```

Comment:
The second argument of `mapping.get` is a default value. Basically, the idea is to use the value of `field.vcoord_type` as the `typeOfLevel` in case there are no entries in the `VCOORD_TYPE` dict for it.